ragna/assistants/_openai_api_compatible.py

```diff
@@ -0,0 +1,74 @@
+import json
+import os
+from typing import AsyncIterator, cast
+
+import httpx
+
+import ragna
+from ragna.core import PackageRequirement, Requirement, Source
+
+from ._api import ApiAssistant
+
+
+class OpenAIApiCompatible(ApiAssistant):
+    def __init__(self) -> None:
+        self._client = httpx.AsyncClient(
+            headers={"User-Agent": f"{ragna.__version__}/{self}"},
+            timeout=60,
+        )
+        self._base_url = os.environ.get("BASE_URL")
+
+    @classmethod
+    def requirements(cls) -> list[Requirement]:
+        return [PackageRequirement("httpx_sse")]
+
+    @classmethod
+    def display_name(cls) -> str:
+        return "OpenAIApiCompatible"
+
+    def _make_system_content(self, sources: list[Source]) -> str:
+        instruction = (
+            "You are an helpful assistants that answers user questions given the context below. "
+            "If you don't know the answer, just say so. Don't try to make up an answer. "
+            "Only use the sources below to generate the answer."
+        )
+        return instruction + "\n\n".join(source.content for source in sources)
+
+    async def _call_api(
+        self, prompt: str, sources: list[Source], *, max_new_tokens: int
+    ) -> AsyncIterator[str]:
+        import httpx_sse
+
+        async with httpx_sse.aconnect_sse(
```

Comment:
Not all APIs use SSE for streaming, e.g. Ollama (#376). Meaning, although the request is the same, the response is coming back in a different protocol and has to be handled accordingly.

Reply:
Given that we already have this logic (work by you!) for the Google assistant, can we just re-use it here?

If so, perhaps just a separate block of code, and another environment variable to control whether this block is executed?